bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfileTest.java

```diff
@@ -670,10 +670,12 @@ public void testComparingInputStateWithItem(GenericItem linkedItem, State inputS
 
     public static Stream<Arguments> testFunctions() {
         NumberItem powerItem = new NumberItem("Number:Power", "powerItem", UNIT_PROVIDER);
+        NumberItem temperatureItem = new NumberItem("Number:Temperature", "temperatureItem", UNIT_PROVIDER);
         NumberItem decimalItem = new NumberItem("decimalItem");
         List<Number> numbers = List.of(1, 2, 3, 4, 5);
         List<Number> negatives = List.of(-1, -2, -3, -4, -5);
         List<QuantityType> quantities = numbers.stream().map(n -> new QuantityType(n, Units.WATT)).toList();
+        List<QuantityType> celsiusQuantities = numbers.stream().map(n -> new QuantityType(n, SIUnits.CELSIUS)).toList();
```

Comment:
Suggested change..
```
List<QuantityType<Temperature>> celsiusQuantities = numbers.stream().map(n -> new <>QuantityType(n, SIUnits.CELSIUS)).toList();
```

Reply:
I've refactored the tests to make it easier.